osu.Game/Online/API/APIAccess.cs

```diff
@@ -143,8 +145,16 @@ private void run()
 
                             failureCount = 0;
 
-                            //we're connected!
-                            state.Value = APIState.Online;
+                            var friendsReq = new GetFriendsRequest();
+                            friendsReq.Success += f =>
+                            {
+                                Friends.AddRange(f);
+
+                                //we're connected!
+                                state.Value = APIState.Online;
```

Comment:
so if the user request succeeds but the friends request fails, the user is no longer going to go online? i don't think that's a good idea.

Reply:
Hmm, not entirely sure how fine it is for a user to not have its friends list retrieved successfully though, that wouldn't make the user fully online to the game?

Reply:
switching to `Failing` sounds like a better solution for sure.

on that note i'd probably split the friends request to a method or something, as nested success callbacks make me uneasy. personal opinion though.